pkg/controller/jobs/job/job_webhook.go

```diff
@@ -70,9 +79,36 @@ func (w *JobWebhook) Default(ctx context.Context, obj runtime.Object) error {
 	log.V(5).Info("Applying defaults", "job", klog.KObj(job))
 
 	jobframework.ApplyDefaultForSuspend(job, w.manageJobsWithoutQueueName)
+
+	if canDefaultManagedBy(job.Spec.ManagedBy) {
```

Comment:
Please cover with unit tests as for JobSet. IIRC we should now be also able to test this e2e by not setting `managedBy` in the test code.